src/System.Collections.Immutable/tests/System.Collections.Immutable.Tests.csproj

```diff
@@ -1,5 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>  
+    <KeepAllProjectReferences>true</KeepAllProjectReferences>  
```

Comment:
nit: If you remove the changes to dir.props and put them in the targets isntead you can move this property with the rest of the properties and not need to import it before dir.props, which in general is not a good idea to try and set things before the dir.props import.